src/android/io/branch/BranchSDK.java

```diff
@@ -558,7 +599,9 @@ private void generateShortUrl(JSONObject options, JSONObject controlParams) thro
             linkProperties.addControlParameter("$windows_phone_url", controlParams.getString("$windows_phone_url"));
         }
 
-        this.branchObj.generateShortUrl(this.activity, linkProperties, new GenerateShortUrlListener());
+        BranchUniversalObject branchObj = (BranchUniversalObject)this.branchObjects.get(instanceIdx);
+
+        branchObj.generateShortUrl(this.activity, linkProperties, new GenerateShortUrlListener());
```

Comment:
We need a listener to call the appropriate callback here as well, so it should take in callbackContext which is lost here but was in PR #74